api-schema/xdb.yaml

```diff
@@ -499,6 +499,12 @@ components:
           $ref: '#/components/schemas/RetryPolicy'
         executeApiRetryPolicy:
           $ref: '#/components/schemas/RetryPolicy'
+        stateFailureRecoveryPolicy:
```

Comment:
WDYT we can put the whole thing under `stateFailureRecoveryPolicy`?
```
stateFailureRecoveryPolicy:
    stateFailureMode: FAIL_PROCESS_ON_STATE_FAILURE or   PROCEED_TO_CONFIGURED_STATE
    stateFailureProceedStateId: xxx  # for FAIL_PROCESS_ON_STATE_FAILURE only 
    stateFailureProceedStateConfig: ... # for FAIL_PROCESS_ON_STATE_FAILURE only
```

Reply:
how about using oneof in state config, and define 2 types of recovery policy?

Reply:
seems not supported, went with your idea

Reply:
yeah, and it's not very well supported in different langauges. that's why I didn't use it at all.